sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/GatewayConnectionConfig.java

```diff
@@ -0,0 +1,259 @@
+     *
+     * @return the default gateway connection configuration.
+     */
+    public static GatewayConnectionConfig getDefaultConfig() {
+        return GatewayConnectionConfig.defaultConfig;
+    }
+
+    /**
+     * Gets the request timeout (time to wait for response from network peer).
+     *
+     * @return the request timeout duration.
+     */
+    public Duration getRequestTimeout() {
+        return this.requestTimeout;
+    }
+
+    /**
+     * Sets the request timeout (time to wait for response from network peer).
+     * The default is 60 seconds.
+     *
+     * @param requestTimeout the request timeout duration.
+     * @return the {@link GatewayConnectionConfig}.
+     */
+    public GatewayConnectionConfig setRequestTimeout(Duration requestTimeout) {
+        this.requestTimeout = requestTimeout;
+        return this;
+    }
+
+    /**
+     * Gets the value of the connection pool size the client is using.
+     *
+     * @return connection pool size.
+     */
+    public int getMaxPoolSize() {
+        return this.maxPoolSize;
+    }
+
+    /**
+     * Sets the value of the connection pool size, the default
+     * is 1000.
+     *
+     * @param maxPoolSize The value of the connection pool size.
+     * @return the {@link GatewayConnectionConfig}.
+     */
+    public GatewayConnectionConfig setMaxPoolSize(int maxPoolSize) {
+        this.maxPoolSize = maxPoolSize;
+        return this;
+    }
+
+    /**
+     * Gets the value of the timeout for an idle connection, the default is 60
+     * seconds.
+     *
+     * @return Idle connection timeout duration.
+     */
+    public Duration getIdleConnectionTimeout() {
+        return this.idleConnectionTimeout;
+    }
+
+    /**
+     * sets the value of the timeout for an idle connection. After that time,
+     * the connection will be automatically closed.
+     *
+     * @param idleConnectionTimeout the duration for an idle connection.
+     * @return the {@link GatewayConnectionConfig}.
+     */
+    public GatewayConnectionConfig setIdleConnectionTimeout(Duration idleConnectionTimeout) {
+        this.idleConnectionTimeout = idleConnectionTimeout;
+        return this;
+    }
+
+    /**
+     * Sets the retry policy options associated with the DocumentClient instance.
+     * <p>
+     * Properties in the RetryOptions class allow application to customize the built-in
+     * retry policies. This property is optional. When it's not set, the SDK uses the
+     * default values for configuring the retry policies.  See RetryOptions class for
+     * more details.
+     *
+     * @param throttlingRetryOptions the RetryOptions instance.
+     * @return the {@link GatewayConnectionConfig}.
+     * @throws IllegalArgumentException thrown if an error occurs
+     */
+    @Override
+    public GatewayConnectionConfig setThrottlingRetryOptions(ThrottlingRetryOptions throttlingRetryOptions) {
+        super.setThrottlingRetryOptions(throttlingRetryOptions);
```

Comment:
why do we need to override the common apis for both gw and direct?

Reply:
Yes, they are common, and they both call their parent's class APIs..
The reason for overriding these ones is because of the return type.
Since our APIs are fluent type, if you directly use parent's API (which returns ConnectionConfig), end users will not be able to use any APIs of DirectConnectionConfig or GatewayConnectionConfig types..

Reply:
This section is no longer valid.